pkg/proc/mem.go

```diff
@@ -136,3 +136,18 @@ func DereferenceMemory(mem MemoryReadWriter) MemoryReadWriter {
 	}
 	return mem
 }
+
+// bufferMemoryReadWriter is dummy a MemoryReadWriter backed by a []byte.
+type bufferMemoryReadWriter struct {
+	buf []byte
+}
+
+func (mem *bufferMemoryReadWriter) ReadMemory(buf []byte, addr uintptr) (n int, err error) {
+	copy(buf, mem.buf[addr-fakeAddress:][:len(buf)])
```

Comment:
Couldn't this be written as `mem.buf[addr-fakeAddress:len(buf)+1]`?

Reply:
That would be wrong, it would have to be `mem.buf [addr-fakeAddress:addr-fakeAddres+len(buf)]`